luatest/cbuilder.lua

```diff
@@ -0,0 +1,231 @@
+--     :use_group('group-001')
+--     :use_replicaset('replicaset-001')
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--
+--     :use_group('group-002')
+--     :use_replicaset('replicaset-002')
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--     :add_instance(<...>)
+--
+--     :config()
+--
+-- The default credentials and iproto options are added to
+-- setup replication and to allow a test to connect to the
+-- instances.
+--
+-- There is a few other methods:
+--
+-- * :set_replicaset_option('foo.bar', value)
+-- * :set_instance_option('instance-001', 'foo.bar', value)
+--
+-- All the methods supports chaining (returns the builder object
+-- back).
+
+local fun = require('fun')
+
+local log = require('luatest.log')
+
+local has_cluster_config, cluster_config = pcall(require, 'internal.config.cluster_config')
+if not has_cluster_config then
+    log.error([[Declarative configuration works on Tarantool 3.0.0+.
+                See tarantool/tarantool@13149d65bc9d for details]])
+end
+
+local base_config = {
+    credentials = {
+        users = {
+            replicator = {
+                password = 'secret',
+                roles = {'replication'},
+            },
+            client = {
+                password = 'secret',
+                roles = {'super'},
+            },
+        },
+    },
+    iproto = {
+        listen = {{
+            uri = 'unix/:./{{ instance_name }}.iproto'
+        }},
+        advertise = {
+            peer = {
+                login = 'replicator',
+            }
+        },
+    },
+    replication = {
+        -- The default value is 1 second. It is good for a real
+        -- usage, but often suboptimal for testing purposes.
+        --
+        -- If an instance can't connect to another instance (say,
+        -- because it is not started yet), it retries the attempt
+        -- after so called 'replication interval', which is equal
+        -- to replication timeout.
+        --
+        -- One second waiting means one more second for a test
+        -- case and, if there are many test cases with a
+        -- replicaset construction, it affects the test timing a
+        -- lot.
+        --
+        -- replication.timeout = 0.1 second reduces the timing
+        -- by half for my test.
+        timeout = 0.1,
+    },
+}
+
+-- Select a group for following calls.
+--
+-- @string group_name Group of test.
+local function cbuilder_use_group(self, group_name)
+    self._group = group_name
+    return self
+end
+
+-- Select a replicaset for following calls.
+--
+-- @string replicaset_name Replica set name.
+local function cbuilder_use_replicaset(self, replicaset_name)
+    self._replicaset = replicaset_name
+    return self
+end
+
+-- Set option to the cluster config.
+--
+-- @string path Option path.
+-- @param value Option value (int, string, table).
+local function cbuilder_set_global_option(self, path, value)
+    assert(type(path) == 'string')
```

Comment:
Is it possible to use checks here and below?

Reply:
I added checks via `checks`.